### PR TITLE
Add the --commit option

### PR DESCRIPTION
**Pull request by guludo**
Hi all.

This PR addresses #38. IMO this feature would be very convenient for users (like me) who do not use zsh and, as such, do not have tab completion set up.

Thank you,
Gustavo Sousa

### CODE REVIEW

**Comment by keis** [comment]
Hi @guludo and thanks for the PR.

Not having to copy-paste is a big reason I wrote this script to begin with so I really would like to make sure it lives up to that. However I'm hesitant to add menu selection code because it either gets very complex or lacking compared to that of other tools.

To tab through a list is (imho) preferable to select by number but this is clearly just preference, I can also imagine some may want to pipe through some other tool like `git fixup | fzf --height '10%'`. Having an option to trigger some external selection tool and commit the result could work though. WDYT?

There's btw the builtin `select` to make numbered selections
```
IFS=$'\n';select commit in $(git fixup); do echo $commit; break; done
```

**Comment by guludo** [comment]
Hi, @keis 

Thank you for the feedback! (Also, thank you for the tip with the `select` builtin).

I like your idea of using an external selection tool.

I updated the last commit to:
1. Allow the user defining the menu command, either via git config or environment variables;
2. Have a fallback menu (intentionally simple) for those users who do not want to depend on an external tool;
3. Document the new features, including a nice example using fzf.

What do you think?

**Comment by keis** [comment]
Cool stuff, especially that fzf example :)

Because there's not really any tests for this script I'll try running it for a bit and see if I notice anything iffy, but in general looks good to me. Do you intend to use the built-in menu or is this now only to have a minimal thing that works?

**Comment by guludo** [comment]
Thanks! :-)

My initial intention was to use the built-in menu, but then I figured out a way to have fzf work together with `git show`. So, now I'm using fzf just the way presented in the README.

So, answering your question, the purpose of the built-in is to serve as minimal thing that works. I think that would be useful for those not willing to pull in extra dependencies in their environment.

**Comment by keis** [comment]
I haven't noticed anything breaking from this so I think it's good to go. I'm not a 100% convinced on the built-in menu but it looks simple enough. If it ends up causing a maintenance burden I might try replacing it with something ruthlessly simple.

File: README.md [hunk 1]
```diff
@@ -70,6 +70,82 @@ $ git rebase -i ...
...
+--commit` display a nice menu with [fzf](https://github.com/junegunn/fzf):
+
+```ini
+[fixup]
+    menu = fzf --height '60%' \
+                --bind 'tab:toggle-preview' \
+                --preview 'git show --color {+1}' \
+                --preview-window=up:80% \
+                --prompt 'Select commit: '
+```
```
Comment (glensc) [inline README.md]: please add this as `git config --global` example

**Comment by glensc** [comment]
the `--commit` option does not work on brew: https://github.com/keis/git-fixup/pull/45

**Pull request merged**